js/packages/mobile-wallet-adapter-protocol/src/types.ts

```diff
@@ -1,192 +1,201 @@
+        Readonly<{
+            max_transactions_per_request: number;
+            max_messages_per_request: number;
+            supported_transaction_versions: ReadonlyArray<TransactionVersion>;
+            features: IdentifierArray;
+            /**
+             * @deprecated Replaced by features array.
+             */
+            supports_clone_authorization: boolean;
+            /**
+             * @deprecated Replaced by features array.
+             */
+            supports_sign_and_send_transactions: boolean;
+        }>
+    >;
+}
+export interface ReauthorizeAPI {
+    reauthorize(params: { auth_token: AuthToken; identity: AppIdentity }): Promise<AuthorizationResult>;
+}
+export interface SignMessagesAPI {
+    signMessages(params: {
+        addresses: Base64EncodedAddress[];
+        payloads: Base64EncodedMessage[];
+    }): Promise<Readonly<{ signed_payloads: Base64EncodedSignedMessage[] }>>;
+}
+export interface SignTransactionsAPI {
+    signTransactions(params: {
+        payloads: Base64EncodedTransaction[];
+    }): Promise<Readonly<{ signed_payloads: Base64EncodedSignedTransaction[] }>>;
+}
+export interface SignAndSendTransactionsAPI {
+    signAndSendTransactions(params: {
+        options?: Readonly<{
+            min_context_slot?: number;
+            commitment?: string;
+            skip_preflight?: boolean;
+            max_retries?: number;
+            wait_for_commitment_to_send_next_transaction?: boolean;
+        }>;
+        payloads: Base64EncodedTransaction[];
+    }): Promise<Readonly<{ signatures: Base64EncodedSignature[] }>>;
+}
+
+export interface MobileWallet
+    extends AuthorizeAPI,
+        CloneAuthorizationAPI,
+        DeauthorizeAPI,
+        GetCapabilitiesAPI,
+        ReauthorizeAPI,
+        SignMessagesAPI,
+        SignTransactionsAPI,
+        SignAndSendTransactionsAPI {}
+
+export interface TerminateSessionAPI {
+    terminateSession(): void
+}
+
+export interface RemoteMobileWallet
+    extends MobileWallet, TerminateSessionAPI {}
+
+// optional features
+export const SolanaSignTransactions = 'solana:signTransactions'
+export const SolanaCloneAuthorization = 'solana:cloneAuthorization'
+export const SolanaSignInWithSolana = 'solana:signInWithSolana'
+
+export type SignInPayload = Readonly<{
+    domain?: string;
+    address?: string;
+    statement?: string;
+    uri?: string;
+    version?: string;
+    chainId?: string;
+    nonce?: string;
+    issuedAt?: string;
+    expirationTime?: string;
+    notBefore?: string;
+    requestId?: string;
+    resources?: readonly string[];
+}> | SolanaSignInInput;
+
+export type SignInPayloadWithRequiredFields = Partial<SignInPayload> & 
+    Required<Pick<SignInPayload, 'domain' | 'address'>>
+
+export type SignInResult = Readonly<{
+    address: Base64EncodedAddress;
+    signed_message: Base64EncodedSignedMessage;
+    signature: Base64EncodedAddress;
+    signature_type?: string;
+}>;
+
+export type Scenario = Readonly<{

```

Comment:
weird diff!

only thing added here are the `Scenario` and `RemtoeScenario` types